test/server_rendered_html_test.rb

```diff
@@ -16,22 +16,55 @@ def wait_to_ensure_asset_pipeline_detects_changes
 
     FileUtils.cp app_file, file_without_updates
     FileUtils.touch app_file
+    wait_to_ensure_asset_pipeline_detects_changes
 
     begin
       get '/server/1'
       refute_match(/Updated/, response.body)
 
-      wait_to_ensure_asset_pipeline_detects_changes
       FileUtils.cp file_with_updates, app_file
       FileUtils.touch app_file
+      wait_to_ensure_asset_pipeline_detects_changes
 
       get '/server/1'
       assert_match(/Updated/, response.body)
     ensure
       # if we have a test failure, we want to make sure that we revert the dummy file
+      FileUtils.mv file_without_updates, app_file
+      FileUtils.touch app_file
       wait_to_ensure_asset_pipeline_detects_changes
+    end
+  end
+
+  test 'react server rendering shows console output as html comment' do
```

Comment:
I think, instead of doing the fragile "copy file dance," this test should just rely on a new route and jsx file in the test app. Even something super specific, like `/console_example`.